test/es-module/test-esm-async-hooks.mjs

```diff
@@ -0,0 +1,47 @@
+// Flags: --experimental-modules
+import '../common/index.mjs';
+import assert from 'assert';
+import { createHook } from 'async_hooks';
+
+const bootstrapCalls = {};
+const calls = {};
+
+const asyncHook = createHook({ init, before, after, destroy, promiseResolve });
+asyncHook.enable();
```

Comment:
Would be good to have a comment here that buffered events will emit synchronously upon hook registration

Reply:
Added.